src/beamlime/core/job.py

```diff
@@ -320,40 +397,144 @@ def reset_job(self, job_id: JobId) -> None:
             status = job.add(data)
             job_statuses.append(status)
+            # Track warnings from job operations, or clear them on success
+            if status.has_error and status.error_message is not None:
+                self._job_warning_messages[job.job_id] = status.error_message
+                self._job_states[job.job_id] = JobState.warning
+            else:
+                # Clear warning state on successful data processing
+                self._job_warning_messages.pop(job.job_id, None)
+                # Only update state if it was warning (preserve error state)
+                if self._job_states.get(job.job_id) == JobState.warning:
+                    self._job_states[job.job_id] = JobState.active
         return job_statuses
 
     def compute_results(self) -> list[JobResult]:
         """
         Compute results from the accumulated data and return them as messages.
         This may include processing the accumulated data and preparing it for output.
         """
-        results = [job.get() for job in self.active_jobs]
+        results = []
+        for job in self.active_jobs:
+            result = job.get()
+            results.append(result)
+            # Track errors from job finalization, or clear them on success
+            if result.error_message is not None:
+                self._job_error_messages[job.job_id] = result.error_message
+                self._job_states[job.job_id] = JobState.error
+            else:
+                # Clear error state on successful finalization
+                self._job_error_messages.pop(job.job_id, None)
+                # Update state based on current status (may still have warnings)
+                if job.job_id in self._job_warning_messages:
+                    self._job_states[job.job_id] = JobState.warning
+                else:
+                    self._job_states[job.job_id] = JobState.active
         self._finish_jobs()
         return results
 
     def _finish_jobs(self):
         for job_id in self._finishing_jobs:
-            _ = self._active_jobs.pop(job_id, None)
+            job = self._active_jobs.pop(job_id, None)
+            if job is not None:
+                self._stopped_jobs[job_id] = job
+                self._job_states[job_id] = JobState.stopped
             _ = self._job_schedules.pop(job_id, None)  # Clean up schedule
         self._finishing_jobs.clear()
 
-    def format_job_error(self, status: JobStatus) -> str:
+    def get_job_status(self, job_id: JobId) -> JobStatus | None:
+        """Get the status of a specific job by its ID."""
+        job = (
+            self._active_jobs.get(job_id)
+            or self._scheduled_jobs.get(job_id)
+            or self._stopped_jobs.get(job_id)
+        )
+        if job is None:
+            return None
+
+        # Determine current state based on job's location in manager
+        if job_id in self._active_jobs:
+            if job_id in self._finishing_jobs:
```

Comment:
We're having to check in various places if the job is in `_finishing_jobs`, or in `_active_jobs`. Can we set the type of job on the job object itself? Would this make things easier?

Maybe it was a way to categorize the jobs so that we don't need to scan all the jobs if we are looking only for fnishing jobs?
Do we expect to have thousands of jobs? If there are only a few jobs running, maybe iterating over all jobs is fast enough?